stellargraph/mapper/sequences.py

```diff
@@ -648,17 +648,12 @@ class CorruptedNodeSequence(Sequence):
     data for training Deep Graph Infomax.
 
     Args:
+        corrupt_inputs: a callable that returns a list of corrupted input values
         base_generator: the uncorrupted Sequence object.
     """
 
-    def __init__(self, base_generator):
-
-        if not isinstance(base_generator, (FullBatchSequence, SparseFullBatchSequence)):
-            raise TypeError(
-                f"base_generator: expected FullBatchSequence or SparseFullBatchSequence, "
-                f"found {type(base_generator).__name__}"
-            )
-
+    def __init__(self, corrupt_inputs, base_generator):
+        self.corrupt_inputs = corrupt_inputs
         self.base_generator = base_generator
         self.targets = np.zeros((1, len(base_generator.target_indices), 2))
```

Comment:
I'm not quite sure how to handle generalising this; potentially a `Generator` class property like:

```python
@classmethod
@abc.abstractmethod
def batch_dimensions(cls):
    ...

class FullBatchGenerator(...):
   @classmethod
   def batch_dimensions(cls):
       return 2

class BatchedNodeGenerator(...):
   @classmethod
   def batch_dimensions(cls):
       return 1
```

which can then be passed through to `__getitem__`:

```python
batch_shape = inputs[0].shape[:batch_dimensions]
targets = np.tile([0, 1], batch_shape + (1,))
```

It's a bit unfortunate that this results in recomputing the target array for each batch, but I guess there's some caching that's possible.

Reply:
I reckon this `Sequence` logic is the trickiest bit. I think it may be worth still having per model code in here because the shuffling/corrupting could be drastically different for different models/training algorithms e.g. graph classification, hinsage, shuffling the adjacency matrix instead of features.

Reply:
or atleast it feels to me that we haven't got enough examples of what this corruption will look like for different algorithms to generalize it yet

Reply:
I think pretty much all options I can think of for shuffling are handled by:

- `corrupt_inputs` being a method on the generator (so it has access to anything in `self`)
- `corrupt_inputs` taking all of the input tensors as input, because it can then pick and choose anything

To be clear, my comment above here is attached to the `self.targets` line and is talking about that, and I don't quite follow how your code connects?

Reply:
ah sorry I didn't realise this was just refferring to the targets line, scratch that comment! 

Just talking about the targets, I think this approach might not work for graph classification. DGI graph classification takes its corrupted nodes from a seperate trainig graph. and so we will likely end up with a different number of true and corrupted nodes. 

Maybe `corrupt_inputs` could be `corrupt_inputs_outputs`?